src/components/TraceQueryBrowser.tsx

```diff
@@ -0,0 +1,214 @@
+import { TextInput, Button } from '@patternfly/react-core';
+import { TraceTable } from './TraceTable';
+import { PersesChartsTheme } from '@perses-dev/components';
+import { Stack, StackItem } from '@patternfly/react-core';
+import { ScatterChart } from '@perses-dev/panels-plugin';
+
+import './TraceQueryBrowser.css';
+
+// To configure a different endpoint for your Tempo instance
+// you need to modify the env variable 'BRIDGE_PLUGIN_PROXY'
+// in the start-console.sh script.
+const proxyDatasource: GlobalDatasource = {
+  kind: 'GlobalDatasource',
+  metadata: { name: 'TempoProxy' },
+  spec: {
+    default: true,
+    plugin: {
+      kind: 'TempoDatasource',
+      spec: {
+        directUrl: '/api/proxy/plugin/distributed-tracing-plugin/backend',
+      },
+    },
+  },
+};
+class DatasourceApiImpl implements DatasourceApi {
+  getDatasource(): Promise<ProjectDatasource | undefined> {
+    return Promise.resolve(undefined);
+  }
+  getGlobalDatasource(): Promise<GlobalDatasource | undefined> {
+    return Promise.resolve(proxyDatasource);
+  }
+  listDatasources(): Promise<ProjectDatasource[]> {
+    return Promise.resolve([]);
+  }
+  listGlobalDatasources(): Promise<GlobalDatasource[]> {
+    return Promise.resolve([proxyDatasource]);
+  }
+  buildProxyUrl(): string {
+    return '/';
+  }
+}
+export const datasourceApi = new DatasourceApiImpl();
+
+export const dashboard = {
+  kind: 'Dashboard',
+  metadata: {},
+  spec: {},
+} as DashboardResource;
+
+// Override eChart defaults with PatternFly colors.
+const patternflyBlue300 = '#2b9af3';
+const patternflyBlue400 = '#0066cc';
+const patternflyBlue500 = '#004080';
+const patternflyBlue600 = '#002952';
+const defaultPaletteColors = [
+  patternflyBlue400,
+  patternflyBlue500,
+  patternflyBlue600,
+];
+const muiTheme = getTheme('light');
+const chartsTheme: PersesChartsTheme = generateChartsTheme(muiTheme, {
+  thresholds: {
+    defaultColor: patternflyBlue300,
+    palette: defaultPaletteColors,
+  },
+});
+
+// PluginRegistry configuration to allow access to
+// visualization panels/charts (@perses-dev/panels-plugin)
+// and data handlers for tempo (@perses-dev/tempo-plugin).
+const pluginLoader = dynamicImportPluginLoader([
+  {
+    resource: panelsResource as PluginModuleResource,
+    importPlugin: () => import('@perses-dev/panels-plugin'),
+  },
+  {
+    resource: tempoResource as PluginModuleResource,
+    importPlugin: () => import('@perses-dev/tempo-plugin'),
+  },
+]);
+
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      refetchOnWindowFocus: false,
+      retry: 0,
+    },
+  },
+});
+
+function TraceQueryBrowser() {
+  const [value, setValue] = React.useState('{}');
+
+  // Use ref to prevent reload on each key tap in TraceQL input box
+  const ref = React.useRef<HTMLInputElement>(null);
+
+  return (
+    <ThemeProvider theme={muiTheme}>
+      <ChartsProvider chartsTheme={chartsTheme}>
+        <SnackbarProvider
```

Comment:
Do we need this snackbar provider?

Reply:
Let me see if I can eliminate it.

Reply:
You're right the snackbar provider was unnecessary. It was removed in the latest push.